lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -39,25 +44,266 @@ struct unexpected_eof : public std::runtime_error {
+            std::replace(value.begin(), value.end(), '\0', ' ');
+        }
+        bool operator()( const NexusData& nd ) const {
+            return this->value == nd.instancename;
+        }
+    };
+    struct varname {
+        std::array< char, 4 > value {{ ' ' }};
+        varname( std::array< char, 4 > x ) : value( x ) {}
+        varname( const std::string& x ) {
+            if (x.size() > value.max_size())
+                throw std::runtime_error("Could not make predicated, gived varname was larger than 8 bytes.");
+            std::copy(x.begin(), x.end(), value.begin());
+            std::replace(value.begin(), value.end(), '\0', ' ');
+        }
+        varname( const char (&x)[9] ) {
+            std::copy(std::begin(x), std::end(x), value.begin());
+            std::replace(value.begin(), value.end(), '\0', ' ');
+        }
+        bool operator()( const NexusData& nd ) const {
+            return this->value == nd.varname;
+        }
+    };
+    struct value {
+        float val;
+        value( float x ) : val( x ) {}
+        bool operator()( const NexusData& nd ) const {
+            return this->val == nd.value;
+        }
+    };
+}
+
+namespace cmp {
+    bool timestep( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.timestep < rhs.timestep;
+    }
+    bool time( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.time < rhs.time;
+    }
+    bool max_perfs( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.max_perfs < rhs.max_perfs;
+    }
+    bool classname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.classname < rhs.classname;
+    }
+    bool instancename( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.instancename < rhs.instancename;
+    }
+    bool varname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.varname < rhs.varname;
+    }
+    bool value( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.value < rhs.value;
+    }
+}
+
+namespace equal {
+    bool timestep( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.timestep == rhs.timestep;
+    }
+    bool time( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.time == rhs.time;
+    }
+    bool max_perfs( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.max_perfs == rhs.max_perfs;
+    }
+    bool classname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.classname == rhs.classname;
+    }
+    bool instancename( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.instancename == rhs.instancename;
+    }
+    bool varname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.varname == rhs.varname;
+    }
+    bool value( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.value == rhs.value;
+    }
+}
+
+struct NexusPlot {
+    NexusHeader header;
+    std::vector< NexusData > data;
```

Comment:
When the `NexusData` is implemented as a `struct` with all public members I think the container should use `const NexusData`

Reply:
I don't think this is actually possible, but by constructing the the plot as `const plt = nex::load(...);`, you essentially achieve the same thing (which is how it's used in all the current code).